From e6b6bd8a4263a36d876ea1b73ee1a12382436756 Mon Sep 17 00:00:00 2001
From: Tanya Singh <tanya_singh@accton.com>
Date: Fri, 1 Aug 2025 09:12:48 +0200
Subject: [PATCH 20/55] base-files: add wlan-ucentral-schema revision hash to
 /etc/openwrt_release

Exposes the uCentral schema version in /etc/openwrt_release, allowing the gateway to verify schema compatibility with deployed access points.

Signed-off-by: Tanya Singh <tanya_singh@accton.com>
---
 include/version.mk                           | 3 +++
 package/base-files/files/etc/openwrt_release | 1 +
 scripts/getver.sh                            | 5 +++++
 3 files changed, 9 insertions(+)

diff --git a/include/version.mk b/include/version.mk
index 3712e844c8..9867f9d2df 100644
--- a/include/version.mk
+++ b/include/version.mk
@@ -61,6 +61,8 @@ VERSION_TIP:=$(shell $(TOPDIR)/scripts/getver.sh wlan-ap)
 VERSION_TIP_VERSION:=$(shell $(TOPDIR)/scripts/getver.sh wlan-ap-version)
 VERSION_TIP_VERSION:=$(if $(VERSION_TIP_VERSION),$(VERSION_TIP_VERSION),devel)
 
+VERSION_UCENTRAL_SCHEMA:=$(shell $(TOPDIR)/scripts/getver.sh ucentral-schema-version)
+
 define taint2sym
 $(CONFIG_$(firstword $(subst :, ,$(subst +,,$(subst -,,$(1))))))
 endef
@@ -112,5 +114,6 @@ VERSION_SED_SCRIPT:=$(SED) 's,%U,$(call sed_escape,$(VERSION_REPO)),g' \
 	-e 's,%h,$(call sed_escape,$(VERSION_HWREV)),g' \
 	-e 's,%a,$(call sed_escape,$(VERSION_TIP)),g' \
 	-e 's,%x,$(call sed_escape,$(VERSION_TIP_VERSION)),g' \
+	-e 's,%B,$(call sed_escape,$(VERSION_UCENTRAL_SCHEMA)),g' \
 	-e 's,%B,$(call sed_escape,$(SOURCE_DATE_EPOCH)),g'
 
diff --git a/package/base-files/files/etc/openwrt_release b/package/base-files/files/etc/openwrt_release
index 3652b1a49a..cbb8f4b820 100644
--- a/package/base-files/files/etc/openwrt_release
+++ b/package/base-files/files/etc/openwrt_release
@@ -7,3 +7,4 @@ DISTRIB_DESCRIPTION='%D %V %C'
 DISTRIB_TAINTS='%t'
 DISTRIB_TIP='%D %V %C / TIP-%x-%a'
 DISTRIB_TIP_VERSION='%x'
+DISTRIB_UCENTRAL_SCHEMA_REVISION='%B'
diff --git a/scripts/getver.sh b/scripts/getver.sh
index c9ae38b91b..fb150f2395 100755
--- a/scripts/getver.sh
+++ b/scripts/getver.sh
@@ -15,6 +15,11 @@ export LC_ALL=C
 	exit 0
 }
 
+[ "$1" = "ucentral-schema-version" ] && {
+    cat ./package/feeds/ucentral/ucentral-schema/Makefile | grep PKG_SOURCE_VERSION | awk -F= '{print substr($2,1,7)}'
+    exit 0
+}
+
 GET_REV=$1
 
 try_version() {
-- 
2.34.1

